Dockerfile.webhook

```diff
@@ -0,0 +1,27 @@
+ARG TAG="release-4.8"
+ARG UBI_IMAGE=registry.access.redhat.com/ubi7/ubi-minimal:latest
+ARG GO_IMAGE=rancher/hardened-build-base:v1.15.8b5
+
+FROM ${GO_IMAGE} as builder
+RUN set -x \
+ && apk --no-cache add \
+    git \
+    make \
+    patch
+ARG TAG
+RUN git clone https://github.com/k8snetworkplumbingwg/sriov-network-operator
+WORKDIR sriov-network-operator
+RUN git fetch --all --tags --prune
+RUN git checkout remotes/origin/${TAG} -b ${TAG}
+COPY 0001-CGO_ENABLED.patch 0002-Change-the-module-URL-from-openshift-to-k8snetworkpl.patch ./
+RUN patch -p1 < 0001-CGO_ENABLED.patch
+RUN patch -p1 < 0002-Change-the-module-URL-from-openshift-to-k8snetworkpl.patch
```

Comment:
I don't really like the fact that we are doing all those `git` and `patch` operations as separate `RUN` instructions, which means that we are creating a layer for each of them. Every single command is creating a new layer which is taking space.

Usually the good practice I like is doing the whole git repo fetching, build, installation and then removing the unneeded source dir in one `RUN` instruction. For example:

```
RUN git clone http://github.com/foo/bar \
    && cd bar \
    && make -j$(proc) \
    && make install \
    && cd .. \
    && rm -rf bar
```

In that way, you are creating just a single filesystem layer with just a binary. Sources are used temporality only for building the binary, but are not bloating the resulting fs layer.

If we've already done something like that in other Rancher images, I think that's a tech debt to fix.

Reply:
That's a great point. Thanks! Let me fix that

Reply:
Reg. sources, note that we are not releasing the image where we build stuff. We are copying the binaries that the ubi image